src/react/projects/spark-core-react/src/SprkCard/SprkCard.js

```diff
@@ -0,0 +1,175 @@
+
+  // Card can either only be base, teaser, or highlighted header
+  warning(
+    onlyHasTeaser || onlyHasHighlightedHeader || hasNoVariant,
+    'SprkCard: cannot have teaserConfig and highlightedHeaderConfig propTypes at the same time.',
+  );
+
+  const GetCardContent = () => {
+    if (onlyHasHighlightedHeader) {
+      return (
+        <SprkCardHighlightedHeader
+          highlightedHeaderConfig={highlightedHeaderConfig}
+        />
+      );
+    }
+
+    if (onlyHasTeaser) {
+      return <SprkCardTeaser teaserConfig={teaserConfig} />;
+    }
+
+    return <React.Fragment>{children}</React.Fragment>;
+  };
+
+  return (
+    <div
+      className={cx('sprk-c-Card sprk-o-Stack', additionalClasses, {
+        'sprk-c-Card--standout': standout,
+        'sprk-u-TextAlign--center': teaserConfig.mediaType === 'icon',
+      })}
+      data-id={idString}
+    >
+      <GetCardContent />
+    </div>
+  );
+};
+
+SprkCard.propTypes = {
+  additionalClasses: PropTypes.string,
+  children: PropTypes.node,
+  highlightedHeaderConfig: PropTypes.shape({
+    // Text inside of highlighted card
+    bodyText: PropTypes.string,
+    // Sub-headline of card
+    description: PropTypes.string,
+    // Headline of card
+    title: PropTypes.string,
+  }),
+  // ID
+  idString: PropTypes.string,
+  // Shadow darker than normal
+  standout: PropTypes.bool,
+
+  // Array of configs for teaserConfig
+  teaserConfig: PropTypes.shape({
+    // Extra classes that on the Call to Action of Teaser
+    additionalCtaClasses: PropTypes.string,
+    // Extra classes that on the Icon of the CTA teaser
+    additionalCtaIconClasses: PropTypes.string,
+    // Add classes to media when it's icon
+    additionalMediaIconClasses: PropTypes.string,
+    // Text inside of teaser card
+    bodyText: PropTypes.string,
+    // Analytics configuration
+    ctaAnalytics: PropTypes.string,
+    // Decides where the Call to Action or the media will go
+    href: PropTypes.string,
+    linkType: PropTypes.oneOf(['anchor', 'router-link']),
+    // Option to add icon next to Call to Action
+    ctaIcon: PropTypes.string,
+    // Text of the Call to Action
+    ctaText: PropTypes.string,
+    // Chooses if the Call ot Action is a link or a button
+    ctaType: PropTypes.oneOf(['link', 'button']),
+    // alt attribute for img
+    mediaAlt: PropTypes.string,
+    // Option for image link analytic tracking
+    mediaLinkAnalytics: PropTypes.string,
+    // Source of main image
+    mediaSrc: PropTypes.string,
+    // Decides where the main media link will go
+    mediaIconName: PropTypes.string,
+    // REQUIRED
+    // Decides if main media of teaser card is an image or an icon
+    mediaType: PropTypes.oneOf(['img', 'icon']),
+    // Teaser's headline
+    title: PropTypes.string,
+    // Will change the order of teaser from title first to main media first
+    titleFirst: PropTypes.bool,
+  }),
+  teaserConfig: PropTypes.shape({
```

Comment:
`teaserConfig` is in here twice. Is one of them supposed to be the highlighted config?